src/plugins/dashboard/public/dashboard_container/embeddable/create/create_dashboard.test.ts

```diff
@@ -59,6 +59,16 @@ test('returns undefined when provided validation function returns redireted', as
   expect(dashboard).toBeUndefined();
 });
 
+test('does not get initial input when provided validation function returns redireted', async () => {
```

Comment:
nit - typo `redireted` => `redirected`

How did loading input cause the filters to disappear?

Reply:
The process was:

1. Short URL gets parsed into Dashboard URL with an `_a` state
2. Dashboard app loads and the Dashboard create

Reply:
so `getInitialInput` modifies the URL, then redirect was getting called. Is that correct?

Reply:
Yes exactly.

Reply:
ok, thanks for the explaination.

There is a bit of a problem with encapsulation, where a create_dashboard test is protecting against a side effect from a callbacks.

I think a clarity in callback name, or test naming, or comments should be added to explain this side effect 

How about one or some of the following
1) rename `getInitialInput` callback to `getInitialInputAndRemoveStateFromUrl` to clarify what callback does and highlight side effect of callback.
2) add comment where `getInitialInput` callback is called that method modifies URL
3) renaming the test case to include the "why". Something like "does not modify URL when validation function returns redirected"

Reply:
The encapsulation concern is completely valid. The reason I left it generic in this way is because the Dashboard container code is usable in many places - most of which have no special URL treatment at all. The method `getInitialInput` can theoretically have any number of side effects.

IMO, in **all** of these cases it is actually more correct for `getInitialInput` to be called _only_ once we're certain that the loaded saved object (if there is one) passes validation. This is why I named the test this way and didn't rename anything to mention the URLs.

Reply:
Thanks for explaining, I had not considered portable dashboards where there is no URL.

How about just a comment in the testing stating that for dashboard application, side effect of getInitialInput removes _a from URL?

Reply:
Great idea. Added a comment in aee1b09eb434f41b8a72948de609f0f5c9b6aecd